External/SPMmodified/xASL/xASL_io_SaveNifti.m

```diff
@@ -229,7 +235,16 @@ function xASL_io_SaveNifti(pathOrigNifti, pathNewNifti, imNew, nBits, bGZip, cha
 %% 4. Remove redundant .mat orientation files
 % Start with removing any pre-existing destination .mat files
 % Even if we don't create a new one, we don't want the wrong .mat motion orientation file with a new NIfTI
-xASL_delete(newMat);
+
+if exist(newMat, 'file')
+    if bOverwrite(3)
+        xASL_delete(newMat);
+    elseif ~bOverwrite(3)
+        % don't do anything
+    else
+        warning('Pre-existing -mat-file')

```

Comment:
@HenkMutsaerts This option will never be reached. As bOverwrite wil be either true or false, but never this third option. If you agree, I'll remove this.

Reply:
Is that true? What about when bOverwrite is a NaN? If we catch those options at the start I'm fine removing this here.

I thought of a way to have a warning only if the user specified something, that doesn't work. But I agree with you we should remove it here and move it to the start

Reply:
Agree: If NaN or Inf is detected, we throw an error. Done in 39d293ee

Of course if user didn't specified anything, then the default values kick in :)

Reply:
Why not simply adding an "else"? You never know that you catch all cases...

But I'm fine otherwise, you can resolve this conversation.

Reply:
The "else" branch contained only "%Do nothing", so no point adding an "else" that has a blank statement.